### PR TITLE
Bump Scala versions

### PR DESCRIPTION
**Pull request by runtologist**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
       fail-fast: false
       matrix:
         java: ['adopt@1.8', 'adopt@1.11']
-        scala: ['2.11.12', '2.12.12', '2.13.3', '0.27.0-RC1']
+        scala: ['2.11.12', '2.12.12', '2.13.3', '3.0.0-M3']
```
Comment (mijicd) [inline .github/workflows/ci.yml:34]: Make sure to update github configuration to include the new check, and remove the old one.

File: sbt [hunk 2]
```diff
@@ -37,7 +37,7 @@ set -o pipefail
 declare -r sbt_release_version="1.3.13"
 declare -r sbt_unreleased_version="1.4.0-M1"
 
-declare -r latest_dotty="0.27.0-RC1"
+declare -r latest_dotty="3.0.0-M3"
```
Comment (mijicd) [inline sbt:40]: I'd update the whole script.

**Review by mijicd — COMMENTED** [review]
Two small comments, other than that should be good to go once the build is passing.

**Comment by runtologist** [comment]
I'm wondering why we see 
```
not found: https://repo1.maven.org/maven2/dev/zio/zio_3.0.0-M3/1.0.3/zio_3.0.0-M3-1.0.3.pom
```
Instead of `3.0.0-M3-1.0.4-2`.

**Comment by mijicd** [comment]
@runtologist Oh, I think you'll have to bump zio version in `build.sbt`.

**Comment by runtologist** [comment]
> @runtologist Oh, I think you'll have to bump zio version in `build.sbt`.
Yes. I've just spent quite a while tryint to figure out why the wrong dependencies were pulled, before I noticed I was editing the wrong file. :roll_eyes:

**Review by mijicd — APPROVED** [review]

**Comment by mijicd** [comment]
@runtologist I've updated the branch protection checks.

**Pull request merged**